openedx/core/djangoapps/user_api/accounts/tests/test_settings_views.py

```diff
@@ -243,3 +246,22 @@ def test_order_history_with_no_product(self):
             order_detail = get_user_orders(self.user)
 
         self.assertEqual(len(order_detail), 1)
+
+    def test_redirect_view(self):
+        with override_waffle_flag(REDIRECT_TO_ACCOUNT_MICROFRONTEND, active=True):
+            old_url_path = reverse('account_settings')
+
+            # Test with waffle flag active, site setting disabled
+            response = self.client.get(path=old_url_path)
+            for attribute in self.FIELDS:
+                self.assertIn(attribute, response.content)
+
+            # Test with waffle flag active, site setting enabled
+            site_domain = 'othersite.example.com'
+            self.set_up_site(site_domain, {
+                'SITE_NAME': site_domain,
+                'ENABLE_ACCOUNT_MICROFRONTEND': True
+            })
+            self.client.login(username=self.USERNAME, password=self.PASSWORD)
+            response = self.client.get(path=old_url_path)
+            self.assertRedirects(response, settings.ACCOUNT_MICROFRONTEND_URL, target_status_code=404)
```

Comment:
FYI: I think you can use `fetch_redirect_response=False` in place of `target_status_code=404`.